include/aws/mqtt/private/client_impl_shared.h

```diff
@@ -10,6 +10,20 @@
 
 struct aws_mqtt_client_connection;
 
+/*
+ * Internal enum that indicates what type of struct the underlying impl pointer actually is.  We use this
+ * to safely interact with private APIs on the implementation or extract the adapted 5 client directly, as
+ * necessary.
+ */
+enum aws_mqtt311_impl_type {
+
+    /* 311 connection impl can be cast to `struct aws_mqtt_client_connection_311_impl` */
+    AWS_MQTT311_IT_311_CONNECTION_IMPL,
```

Comment:
Trivial/Debatable: Can we remove the "IMPL" from the backs or remove "IT" from the mid of these enums? I think it'd make more sense to remove the "IMPL" at the end to follow the `aws_mqtt311_impl_type` name. e.g. "AWS_MQTT311_IT_311_CONNECTION"